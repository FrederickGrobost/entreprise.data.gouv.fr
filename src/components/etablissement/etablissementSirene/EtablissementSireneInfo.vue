<template>
  <div class="company__panel panel">
    <div class="name__category">
      <h4>Informations générales</h4>
    </div>
    <div class="company__item"><label class="company__item-key">SIRET</label><div class="company__item-value"> {{ resultSirene.siret | prettySiret }}</div></div>
    <div class="company__item"><label class="company__item-key">SIREN</label><div class="company__item-value"> {{ resultSirene.siren | prettySiren }}</div></div>
    <div class="company__item"><label class="company__item-key">Clef NIC</label><div class="company__item-value"> {{ resultSirene.nic }}</div></div>
    <div class="company__item"><label class="company__item-key">Activité principale Etablissement</label><div class="company__item-value"> {{ resultSirene.activite_principale }} - {{ resultSirene.libelle_activite_principale }}</div></div>
    <div class="company__item"><label class="company__item-key">Activité principale Entreprise</label><div class="company__item-value"> {{ resultSirene.activite_principale_entreprise }} - {{ resultSirene.libelle_activite_principale_entreprise }}</div></div>
    <div class="company__item"><label class="company__item-key">Nature juridique</label><div class="company__item-value"> {{ resultSirene.libelle_nature_juridique_entreprise | ifExist }}</div></div>
    <div class="company__item"><label class="company__item-key">Numéro RNA</label><div class="company__item-value"> {{ resultSirene.numero_rna | ifExist }}</div></div>
    <div class="company__item"><label class="company__item-key">N° TVA Intracommunautaire</label><div class="company__item-value"> {{ tvaIntracommunautaire | prettyTvaNumber }}</div></div>
  </div>
</template>

<script>
import Filters from '@/components/mixins/filters'

export default {
  name: 'EtablissementSireneInfo',
  computed: {
    resultSirene () {
      return this.$store.getters.singlePageEtablissementSirene
    },
    tvaIntracommunautaire () {
      const tvaKey = (12 + 3 * (this.resultSirene.siren % 97)) % 97
      const tvaNumber = `${tvaKey}${this.resultSirene.siren}`
      return `FR${tvaNumber}`
    }
  },
  mixins: [Filters]
}
</script>
