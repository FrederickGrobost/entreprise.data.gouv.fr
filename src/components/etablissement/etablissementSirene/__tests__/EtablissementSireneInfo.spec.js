import EtablissementSireneInfo from "@/components/etablissement/etablissementSirene/EtablissementSireneInfo.vue";
import { createLocalVue, mount } from "@vue/test-utils";
import { __createMocks as createStoreMocks } from "@/store/index.js";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);
jest.mock("@/store/index.js");

describe("EtablissementSireneInfo.vue", () => {
  // let storeMocks
  // let wrapperEPI
  // let etablissementPanelInfo
  // beforeEach(() => {
  //   storeMocks = createStoreMocks()
  //   wrapperEPI = shallowMount(EtablissementSireneInfo, {
  //     localVue,
  //     store: storeMocks.store
  //   })
  //   etablissementPanelInfo = wrapperEPI.vm
  // })
  // TODO
  // test('tvaIntracommunautaire) is correctly computed', () => {
  //   expect(etablissementPanelInfo.tvaIntracommunautaire).toBe('FR80833057201')
  // })
});

describe("EtablissementSireneInfo.vue snapshot", () => {
  const localVue = createLocalVue();
  const storeMocks = createStoreMocks();
  const wrapperEPI = mount(EtablissementSireneInfo, {
    localVue,
    store: storeMocks.store
  });

  test("It match the snapshot", () => {
    expect(wrapperEPI.vm.$el).toMatchSnapshot();
  });
});
