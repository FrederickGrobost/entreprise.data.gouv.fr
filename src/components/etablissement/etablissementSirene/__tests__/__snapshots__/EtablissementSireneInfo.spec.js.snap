// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementSireneInfo.vue snapshot It match the snapshot 1`] = `
<div
  class="company__panel panel"
>
  <div
    class="name__category"
  >
    <h4>
      Informations générales
    </h4>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      SIRET
    </label>
     
    <div
      class="company__item-value"
    >
      
      000 000 000 00001
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      SIREN
    </label>
     
    <div
      class="company__item-value"
    >
      
      
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Clef NIC
    </label>
     
    <div
      class="company__item-value"
    >
      
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Activité principale Etablissement
    </label>
     
    <div
      class="company__item-value"
    >
      
      62.0Z -
      
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Activité principale Entreprise
    </label>
     
    <div
      class="company__item-value"
    >
      
      01.21Z -
      Culture de la vigne
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Nature juridique
    </label>
     
    <div
      class="company__item-value"
    >
      
      1000 -
      Entrepreneur individuel
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      Numéro RNA
    </label>
     
    <div
      class="company__item-value"
    >
      
      Non renseigné
    
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <label
      class="company__item-key"
    >
      N° TVA Intracommunautaire
    </label>
     
    <div
      class="company__item-value"
    >
      
      FRNaNundefined
    
    </div>
  </div>
</div>
`;
